test/add-member.test.js

```diff
@@ -272,3 +273,262 @@ test('addMembers too many members', async (t) => {
+    )
+      .then((res) => {
+        t.pass('got peer roots')
+        return res
+      })
+      .catch((err) => t.error(err))
+  ).map((root) => root.id)
+
+  await Promise.all([
+    replicate(alice, bob),
+    replicate(alice, carol),
+    replicate(alice, david),
+  ])
+    .then(() => t.pass('replicated'))
+    .catch((err) => t.error(err))
+
+  const { id: groupId, writeKey: firstEpochKey } = await alice.tribes2
+    .create()
+    .then((res) => {
+      t.pass('alice created group')
+      return res
+    })
+    .catch((err) => t.error(err))
+  const firstEpochSecret = firstEpochKey.key.toString('base64')
+
+  const { key: firstEpochPostId } = await alice.tribes2
+    .publish({
+      type: 'test',
+      text: 'first post',
+      recps: [groupId],
+    })
+    .then((res) => {
+      t.pass('alice published in first epoch')
+      return res
+    })
+    .catch((err) => t.error(err))
+
+  await alice.tribes2
+    .addMembers(groupId, [bobRootId, carolRootId])
+    .then(() => t.pass('alice added bob and carol'))
+    .catch((err) => t.error(err))
+
+  await Promise.all([
+    replicate(alice, bob),
+    replicate(alice, carol),
+    replicate(alice, david),
+  ])
+    .then(() => t.pass('replicated'))
+    .catch((err) => t.error(err))
+
+  await bob.tribes2
+    .acceptInvite(groupId)
+    .then(() => t.pass('bob accepted invite'))
+    .catch((err) => t.error(err))
+
+  await Promise.all([
+    alice.tribes2.excludeMembers(groupId, [carolRootId]),
+    bob.tribes2.excludeMembers(groupId, [carolRootId]),
+  ])
+    .then(() => t.pass('alice and bob excluded carol'))
+    .catch((err) => t.error(err))
+
+  const { key: aliceForkPostId } = await alice.tribes2
+    .publish({
+      type: 'test',
+      text: 'alice fork post',
+      recps: [groupId],
+    })
+    .then((res) => {
+      t.pass('alice published in her fork')
+      return res
+    })
+    .catch((err) => t.error(err))
+  const { writeKey: aliceForkKey } = await alice.tribes2
+    .get(groupId)
+    .then((res) => {
+      t.pass('alice got info on her fork')
+      return res
+    })
+    .catch((err) => t.error(err))
+  const aliceForkSecret = aliceForkKey.key.toString('base64')
+
+  const { key: bobForkPostId } = await bob.tribes2
+    .publish({
+      type: 'test',
+      text: 'bob fork post',
+      recps: [groupId],
+    })
+    .then((res) => {
+      t.pass('bob posted in his fork')
+      return res
+    })
+    .catch((err) => t.error(err))
+  const { writeKey: bobForkKey } = await bob.tribes2
+    .get(groupId)
+    .then((res) => {
+      t.pass('bob got info on his fork')
+      return res
+    })
+    .catch((err) => t.error(err))
+  const bobForkSecret = bobForkKey.key.toString('base64')
+
+  await Promise.all([
+    replicate(alice, bob),
+    replicate(alice, carol),
+    replicate(alice, david),
+  ])
+    .then(() => t.pass('replicated'))
+    .catch((err) => t.error(err))
```

Comment:
Locally it logs errors when running this, but things still pass

```
addMembers adds to all the tip epochs and gives keys to all the old epochs as well
    ✔ clients started
    ✔ got peer roots
    ✔ replicated
    ✔ alice created group
    ✔ alice published in first epoch
    ✔ alice added bob and carol
    ✔ replicated
    ✔ bob accepted invite
    ✔ alice and bob excluded carol
    ✔ alice published in her fork
    ✔ alice got info on her fork
    ✔ bob posted in his fork
    ✔ bob got info on his fork
Error: fatal error getting preferred epoch
    at /home/me/prj/ssb/ssb-tribes2/listeners.js:144:52
    at buildPreferredEpoch (/home/me/prj/ssb/ssb-tribes2/lib/epochs.js:552:14)
  Error: Tips do not have same membership - case not handled yet
    at buildPreferredEpoch (/home/me/prj/ssb/ssb-tribes2/lib/epochs.js:553:9)
    at /home/me/prj/ssb/ssb-tribes2/lib/epochs.js:256:7
    at /home/me/prj/ssb/ssb-tribes2/lib/epochs.js:380:9
    at /home/me/prj/ssb/ssb-tribes2/node_modules/.pnpm/pull-stream@3.7.0/node_modules/pull-stream/sinks/reduce.js:10:5
    at /home/me/prj/ssb/ssb-tribes2/node_modules/.pnpm/pull-stream@3.7.0/node_modules/pull-stream/sinks/drain.js:26:24
    at /home/me/prj/ssb/ssb-tribes2/node_modules/.pnpm/pull-stream@3.7.0/node_modules/pull-stream/throughs/async-map.js:28:19
    at /home/me/prj/ssb/ssb-tribes2/node_modules/.pnpm/pull-stream@3.7.0/node_modules/pull-stream/throughs/flatten.js:35:20
    at /home/me/prj/ssb/ssb-tribes2/node_modules/.pnpm/pull-stream@3.7.0/node_modules/pull-stream/sources/values.js:19:7
    at nextStream (/home/me/prj/ssb/ssb-tribes2/node_modules/.pnpm/pull-stream@3.7.0/node_modules/pull-stream/throughs/flatten.js:33:9)
    at /home/me/prj/ssb/ssb-tribes2/node_modules/.pnpm/pull-stream@3.7.0/node_modules/pull-stream/throughs/flatten.js:22:31
    ✔ replicated
    ✔ david got added to the group by alice
Error: fatal error getting preferred epoch
    at /home/me/prj/ssb/ssb-tribes2/listeners.js:144:52
    at buildPreferredEpoch (/home/me/prj/ssb/ssb-tribes2/lib/epochs.js:552:14)
  Error: Tips do not have same membership - case not handled yet
    at buildPreferredEpoch (/home/me/prj/ssb/ssb-tribes2/lib/epochs.js:553:9)
    at /home/me/prj/ssb/ssb-tribes2/lib/epochs.js:256:7
    at /home/me/prj/ssb/ssb-tribes2/lib/epochs.js:380:9
    at /home/me/prj/ssb/ssb-tribes2/node_modules/.pnpm/pull-stream@3.7.0/node_modules/pull-stream/sinks/reduce.js:10:5
    at /home/me/prj/ssb/ssb-tribes2/node_modules/.pnpm/pull-stream@3.7.0/node_modules/pull-stream/sinks/drain.js:26:24
    at /home/me/prj/ssb/ssb-tribes2/node_modules/.pnpm/pull-stream@3.7.0/node_modules/pull-stream/throughs/async-map.js:28:19
    at /home/me/prj/ssb/ssb-tribes2/node_modules/.pnpm/pull-stream@3.7.0/node_modules/pull-stream/throughs/flatten.js:35:20
    at /home/me/prj/ssb/ssb-tribes2/node_modules/.pnpm/pull-stream@3.7.0/node_modules/pull-stream/sources/values.js:19:7
    at nextStream (/home/me/prj/ssb/ssb-tribes2/node_modules/.pnpm/pull-stream@3.7.0/node_modules/pull-stream/throughs/flatten.js:33:9)
    at /home/me/prj/ssb/ssb-tribes2/node_modules/.pnpm/pull-stream@3.7.0/node_modules/pull-stream/throughs/flatten.js:22:31
    ✔ David got added to both forks
    ✔ alice got her additions of david
    ✔ gave david the secret to alice's fork
    ✔ gave david the secret to the initial epoch, in the addition to alice's fork
    ✔ gave david the secret to bob's fork
    ✔ gave david the secret to the initial epoch, in the addition to bob's fork
    ✔ replicated
    ✔ david accepted invite
    ✔ david got bob's post in his fork
    ✔ david decrypted the msg in bob's fork
    ✔ david got alice's post in her fork
    ✔ david decrypted the msg in alice's fork
    ✔ david got alice's post in the initial epoch
    ✔ david decrypted the msg in the first epoch
    ✔ clients got closed

  total:     30
  passing:   30
  18.914 s
```

but in CI we get tape errors here

```
addMembers adds to all the tip epochs and gives keys to all the old epochs as well
    ✔ clients started
    ✔ got peer roots
    ✔ replicated
    ✔ alice created group
    ✔ alice published in first epoch
    ✔ alice added bob and carol
    ✔ replicated
    ✔ bob accepted invite
    ✔ alice and bob excluded carol
    ✔ alice published in her fork
    ✔ alice got info on her fork
    ✔ bob posted in his fork
    ✔ bob got info on his fork
    ✖ 39) Error: retryUntil timed out
      Expected error to be falsy
      At: <anonymous> (/test/add-member.test.js:431:23)
      
    ✔ david got added to the group by alice
    ✖ 41) David got added to both forks
      Expected 2 but got 1
      At: Test.<anonymous> (/test/add-member.test.js:441:5)
      
    ✔ alice got her additions of david
    ✔ gave david the secret to alice's fork
    ✔ gave david the secret to the initial epoch, in the addition to alice's fork
    ✖ 45) TypeError: Cannot read properties of undefined (reading 'secret')
      Expected error to be falsy
```

Reply:
started using `peer.getVectorClock` to get even more feeds to replicate since we seemed to be missing some. replication seems to work better now but now addMembers breaks in a lot of tests. are we running into https://github.com/ssbc/ssb-box2/issues/30 ? how?? i'm going crazy

Reply:
fixed it (i think) in a call with andre. i think the problem was that alice can pass her secret group feed to bob, but bob can't pass it to carol, at least using branchStream since that's based on what he can decrypt. yet bob<->carol fails because bob has an extra feed that shows up in the clock. so the fix was to directly replicate from alice to the others since she's the only one that can give people her (so far) secret feed.